### ISSUE TITLE
Processing model for off-the-web Web Book?

### ISSUE THREAD

**Issue by rdeltour** [therealglazou/webbook#12]
I believe the spec is not very explicit on how a UA is supposed to process a Web Book.

For instance, to load a Web Book, would a UA unzip it first? then load its constituants as `file` URLs? or another fetch scheme? or something else?

What's the origin of a Web Book?

Various features of the Web depends on this to be further defined, e.g. if authors want to start using SW on a Web Book.

This is related to #6 and #9.

**Comment by therealglazou** [comment]
Good comment, I plan to make the spec focus more on the UA requirements (should, must, recommend, etc). That said, very naive question: do we HAVE to resolve that or would it be possible to leave it up to the UA? I don't have the answer yet, but I feel the question is worth a discussion.

**Comment by rdeltour** [comment]
> That said, very naive question: do we HAVE to resolve that or would it be possible to leave it up to the UA?

Good question. The way I see it, leaving it up to the UA means that the processing model may not be consistent, and authors who depend on it will have to cope with interoperability issues. In a way, it's not respecting the priority of constituencies.

**Comment by therealglazou** [comment]
> Good question. The way I see it, leaving it up to the UA means that the processing model may not be consistent, and authors who depend on it will have to cope with interoperability issues. In a way, it's not respecting the priority of constituencies.

Right, but let's take the example of ZIP vs. WebPackage. I chose ZIP, just to remain compatible with EPUB3 but why not WebPackage in the long run? Why enforce that as soon as it is possible to recognize the packaging mechanism from, for instance, magic bytes or HTTP headers? I think it's a real question.
